server/assets-v1/templates/src/pages/user_portal/password_reset/reset-password-page.html

```diff
@@ -1,178 +1,205 @@
-            alert("Repeated password does not match");
-            return;
-        }
+                alert("All fields are mandatory");
+                return;
+            }
+            if (newPassword.value !== repeatedNewPassword.value) {
+                alert("Repeated password does not match");
+                return;
+            }
 
-        let currMnemonic = mnemonicSentence.value.trim();
+            let currMnemonic = mnemonicSentence.value.trim();
 
-        if (!mnemonic.isValid(currMnemonic)) {
-            alert("The provided mnemonic is invalid");
-            return;
-        }
+            if (!mnemonic.isValid(currMnemonic)) {
+                alert("The provided mnemonic is invalid");
+                return;
+            }
 
-        const keyPair = mnemonic.getPrivateAndPublicKeys(currMnemonic);
-        const signature = mnemonic.sign(keyPair.privateKey, messageToSign);
+            const keyPair = mnemonic.getPrivateAndPublicKeys(currMnemonic);
+            const signature = mnemonic.sign(keyPair.privateKey, messageToSign);
 
-        try {
-            const response = await window.fetch(
-                    "[[ .ProxyURL ]]/api/v1/reset-password",
+            try {
+                const responseOne = await window.fetch(
+                    "[[ .ProxyURL ]]/api/v2/reset-password-precheck",
                     {
                         method: "POST",
                         headers: {
                             "Content-Type": "application/json",
                         },
                         body: JSON.stringify({
                             username: username.value,
-                            signature: Array.from(signature),
-                            new_password: newPassword.value
                         }),
                     },
-            );
+                );
 
-            const result = await response.json();
+                const resetPasswordPrecheckResponseBody = await responseOne.json();
+                if (responseOne.status !== 200) {
+                    alert("Something wen wrong!");
+                    return;
+                }
 
-            if (response.status === 201) {
-                alert(result.message);
-                window.location.href = "[[ .ProxyURL ]]/user-login-page";
-            } else {
-                console.log(result.errors);
-                alert("Error: " + result.message);
+                const saltedPassword = CryptoJS.PBKDF2(
+                    newPassword.value,
+                    CryptoJS.enc.Hex.parse(resetPasswordPrecheckResponseBody.data.salt),
+                    {
+                        keySize: 160 / 32,
+                        iterations: resetPasswordPrecheckResponseBody.data.iterationCount,
+                        hasher: CryptoJS.algo.SHA1,
+                    }
+                ).toString(CryptoJS.enc.Hex);
+
+                const CLIENT_KEY_ENV = "TEST_CLIENT_KEY";
```

Comment:
The correct value for client key is `"Client Key"`, you can also inline it into line 149 (since it's used only once). Please check register_v2.html to see the correct way of generating hashes.